File: OneCodeTeam/Load-JavaScript-Resources-on-demand-16100/[C#]-Load JavaScript Resources on-demand (CSASPNETLoadAndExecuteDynamicJS)/C#/CSASPNETLoadAndExecuteDynamicJS/Service1.svc.cs
/****************************** Module Header ******************************\
 * Module Name: Service1.svc.cs
 * Project:     CSASPNETLoadAndExecuteDynamicJS
 * Copyright (c) Microsoft Corporation. 
 * This source is subject to the Microsoft Public License. 
 *
 * A simple service.
 * 
 * See http://www.microsoft.com/en-us/openness/resources/licenses.aspx#MPL 
 * All other rights reserved. 
 * 
 * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
 * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
 * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\****************************************************************************/
using System;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.ServiceModel.Activation;
using System.ServiceModel.Web;
using System.Text;
using System.Web.Services;
using System.Web.Script.Services;
using System.Web;

namespace CSASPNETLoadAndExecuteDynamicJS
{
    [ServiceContract(Namespace = "")]
    [AspNetCompatibilityRequirements(RequirementsMode = AspNetCompatibilityRequirementsMode.Allowed)]
    public class Service1
    {
        // To use HTTP GET, add [WebGet] attribute. (Default ResponseFormat is WebMessageFormat.Json)
        // To create an operation that returns XML,
        //     add [WebGet(ResponseFormat=WebMessageFormat.Xml)],
        //     and include the following line in the operation body:
        //         WebOperationContext.Current.OutgoingResponse.ContentType = "text/xml";
        [OperationContract]
        [WebMethod]
        [ScriptMethod(ResponseFormat = ResponseFormat.Json)]
        public string LoadScript(string source)
        {
            // Add your operation implementation here
            return "Scripts/" + source;
        }

        // Add more operations here and mark them with [OperationContract]
    }
}

File: OneCodeTeam/Load-javascript-resources-on-demand-16490/[VB]-Load javascript resources on demand (VBASPNETLoadAndExecuteDynamicJS)/VB/VBASPNETLoadAndExecuteDynamicJS/Service1.svc.vb
'****************************** Module Header ******************************\
' Module Name:  Service1.svc.vb
' Project:      VBLoadAndExecuteDynamicJS
' Copyright (c) Microsoft Corporation
'
' This is the AJAX-enabled WCF Service
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************/

Imports System.ServiceModel
Imports System.ServiceModel.Activation
Imports System.Web.Script.Services
Imports System.Web.Services

<ServiceContract(Namespace:="")> _
<AspNetCompatibilityRequirements(RequirementsMode:=AspNetCompatibilityRequirementsMode.Allowed)> _
Public Class Service1

    ' To use HTTP GET, add [WebGet] attribute. (Default ResponseFormat is WebMessageFormat.Json)
    ' To create an operation that returns XML,
    '     add [WebGet(ResponseFormat=WebMessageFormat.Xml)],
    '     and include the following line in the operation body:
    '         WebOperationContext.Current.OutgoingResponse.ContentType = "text/xml";
    <OperationContract()> _
    <WebMethod()> _
    <ScriptMethod(ResponseFormat:=ResponseFormat.Json)> _
    Public Function LoadScript(ByVal source As String) As String
        ' Add your operation implementation here
        Return "Scripts/" & source
    End Function

    ' Add more operations here and mark them with <OperationContract()>

End Class
